library/java/test/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt

```diff
@@ -0,0 +1,33 @@
+package io.envoyproxy.envoymobile.engine
+
+import org.junit.Test
+
+private const val TEST_CONFIG = """
+mock_template:
+- name: mock
+  connect_timeout: {{ connect_timeout }}
+  dns_refresh_rate: {{ dns_refresh_rate }}
+  stats_flush_interval: {{ stats_flush_interval }}
+"""
+
+
+class EnvoyConfigurationTest {
+
+  @Test
+  fun `resolving with default configuration resolves with values`() {
+    val envoyConfiguration = EnvoyConfiguration(123, 234, 345)
+
+    val resolvedTemplate = envoyConfiguration.resolveTemplate(TEST_CONFIG)
+    assertThat(resolvedTemplate).contains("connect_timeout: 123s")
+    assertThat(resolvedTemplate).contains("dns_refresh_rate: 234s")
+    assertThat(resolvedTemplate).contains("stats_flush_interval: 345s")
```

Comment:
Maybe also assert that the template doesn't contain any `{{`?

Reply:
The behavior that is baked in the method is that if it contains any unresolved templates, we'll throw an exception so that test is: `resolve templates with invalid templates will throw on build`